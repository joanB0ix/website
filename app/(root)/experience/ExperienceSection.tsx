import { Section } from "@/components/Section";
import { Experience } from "@/config/experienceList";

export interface ExperienceSectionProps {
  experienceList: Experience[];
}

export default function ExperienceSection({
  experienceList,
}: ExperienceSectionProps) {
  return (
    <Section title="Experience">
      <div className="flex flex-col gap-y-2">
        {experienceList.map((experience) => {
          return (
            <div key={experience.start_month}>
              <div className="flex flex-col sm:flex-row justify-between mb-1">
                <div className="flex flex-col">
                  <h3 className="font-medium">{experience.company}</h3>
                  <span>{experience.position}</span>
                  <div className="block sm:hidden mt-1">
                    <div>{experience.location}</div>
                    <div>
                      {experience.start_month} {experience.start_year},{" "}
                      {experience.end_month
                        ? experience.end_month + " " + experience.end_year
                        : "Present"}
                    </div>
                  </div>
                </div>
                <div className="hidden sm:flex flex-col text-right">
                  <div>{experience.location}</div>
                  <div>
                    {experience.start_month} {experience.start_year},{" "}
                    {experience.end_month
                      ? experience.end_month + " " + experience.end_year
                      : "Present"}
                  </div>
                </div>
              </div>
              <ul className="list-disc pl-6">
                {experience.description.map((bullet) => {
                  return (
                    <li className="text-pretty" key={bullet}>
                      <p>{bullet}</p>
                    </li>
                  );
                })}
              </ul>
            </div>
          );
        })}
      </div>
    </Section>
  );
}
