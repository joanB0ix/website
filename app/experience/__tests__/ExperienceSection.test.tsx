import { render, screen } from "@testing-library/react";
import ExperienceSection, {
  ExperienceSectionProps,
} from "../ExperienceSection";

describe("ExperienceSection", () => {
  const mockExperienceList: ExperienceSectionProps["experienceList"] = [
    {
      company: "Company One",
      position: "Software Engineer",
      location: "New York, NY",
      start_month: "January",
      start_year: 2020,
      end_month: "March",
      end_year: 2022,
      description: [
        "Developed web applications",
        "Implemented CI/CD pipelines",
      ],
    },
    {
      company: "Company Two",
      position: "Senior Developer",
      location: "San Francisco, CA",
      start_month: "April",
      start_year: 2022,
      description: ["Led team of developers", "Architected scalable systems"],
    },
  ];

  it("Should render correctly", () => {
    render(<ExperienceSection experienceList={mockExperienceList} />);

    const sectionTitle = screen.getByText("Experience");
    expect(sectionTitle).toBeInTheDocument();

    mockExperienceList.forEach((experience) => {
      const companyName = screen.getByText(experience.company);
      expect(companyName).toBeInTheDocument();

      const position = screen.getByText(experience.position);
      expect(position).toBeInTheDocument();

      const location = screen.getAllByText(experience.location);
      expect(location[0]).toBeInTheDocument();

      const dateRange = screen.getAllByText(
        experience.end_month
          ? `${experience.start_month} ${experience.start_year}, ${experience.end_month} ${experience.end_year}`
          : `${experience.start_month} ${experience.start_year}, Present`
      );
      expect(dateRange[0]).toBeInTheDocument();

      experience.description.forEach((bullet) => {
        const bulletText = screen.getByText(bullet);
        expect(bulletText).toBeInTheDocument();
      });
    });
  });

  it("Should not render list when there are no experiences", () => {
    render(<ExperienceSection experienceList={[]} />);
    const sectionContent = screen.getByText("Experience");
    expect(sectionContent).toBeInTheDocument();

    const items = screen.queryAllByRole("listitem");
    expect(items).toHaveLength(0);
  });

  it("Should match snapshot", () => {
    const { asFragment } = render(
      <ExperienceSection experienceList={mockExperienceList} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
