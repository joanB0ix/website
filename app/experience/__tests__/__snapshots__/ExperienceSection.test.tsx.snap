// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExperienceSection Should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-6"
  >
    <h2
      class="font-bold pb-2"
    >
      Experience
    </h2>
    <div
      class="flex flex-col gap-y-2"
    >
      <div>
        <div
          class="flex justify-between mb-1"
        >
          <div
            class="flex flex-col"
          >
            <h3
              class="font-medium"
            >
              Company One
            </h3>
            <span>
              Software Engineer
            </span>
          </div>
          <div
            class="flex flex-col text-right"
          >
            <div>
              New York, NY
            </div>
            <div>
              January 2020, March 2022
            </div>
          </div>
        </div>
        <ul
          class="list-disc pl-6"
        >
          <li
            class="text-pretty"
          >
            <p>
              Developed web applications
            </p>
          </li>
          <li
            class="text-pretty"
          >
            <p>
              Implemented CI/CD pipelines
            </p>
          </li>
        </ul>
      </div>
      <div>
        <div
          class="flex justify-between mb-1"
        >
          <div
            class="flex flex-col"
          >
            <h3
              class="font-medium"
            >
              Company Two
            </h3>
            <span>
              Senior Developer
            </span>
          </div>
          <div
            class="flex flex-col text-right"
          >
            <div>
              San Francisco, CA
            </div>
            <div>
              April 2022, Present
            </div>
          </div>
        </div>
        <ul
          class="list-disc pl-6"
        >
          <li
            class="text-pretty"
          >
            <p>
              Led team of developers
            </p>
          </li>
          <li
            class="text-pretty"
          >
            <p>
              Architected scalable systems
            </p>
          </li>
        </ul>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
